plugins/grafana-custom-plugins/grafana-dependency-plugin/src/DependencyPanel.tsx

```diff
@@ -44,26 +61,45 @@ export const DependencyPanel: React.FC<Props> = ({ options, data, width, height
 
   for (let i = 0; i < frame.length; i++) {
     const sourcePodName = sourcePodNames?.values.get(i);
+    const sourcePodLabel = sourcePodLabels?.values.get(i);
     const sourceNodeName = sourceNodeNames?.values.get(i);
     const destinationPodName = destinationPodNames?.values.get(i);
+    const destinationPodLabel = destinationPodLabels?.values.get(i);
     const destinationNodeName = destinationNodeNames?.values.get(i);
     const destinationServicePortName = destinationServicePortNames?.values.get(i);
     const octetDeltaCount = octetDeltaCounts?.values.get(i);
 
+    function getName(groupByLabel: boolean, source: boolean, labelJSON: string) {
+      if(!groupByLabel || labelJSON === undefined || options.labelName === undefined) {
+        return source ? sourcePodName : destinationPodName;
+      }
+      let labels = JSON.parse(labelJSON);
+      if(labels[options.labelName] !== undefined) {
+        return labels[options.labelName];
+      }
+      return sourcePodName;
+    }
+
+    // TODO add button or switch to set groupByLabelVariable
```

Comment:
little confused on the TODO comment here. What is it for? I thought we already have a switch in the panel options?

Reply:
I meant to delete that earlier